src/classLibrary/Executor.ts

```diff
@@ -0,0 +1,148 @@
+import { GlobFactory } from "./Utility/GlobFactory";
+import chalk from "chalk";
+import { FileFactory } from "./Utility/FileFactory";
+import deleteEmpty from "delete-empty";
+import { PluginHandler } from "./PluginHandler";
+
+class Executor {
+    private readonly util: Utility;
+    private guidGenerator: IGuidGenerator;
+
+    constructor(util: Utility, guidGenerator: IGuidGenerator) {
+        this.util = util;
+        this.guidGenerator = guidGenerator;
+    }
+    
+    async GenerateVFS(cyanSafe: CyanSafe, globFactory: GlobFactory): Promise<VirtualFileSystemInstance[]> {
+    
+        //package resolver may need to think how to expand it to beyond npm modules
+        let strategies: IParsingStrategy[] = [
+            new GuidResolver(this.guidGenerator, this.util),
+            new IfElseResolver(this.util),
+            new InlineFlagResolver(this.util),
+            new VariableResolver(this.util)
+        ];
+
+        let parser: Parser = new Parser(this.util, strategies, cyanSafe);
+
+        console.log(chalk.greenBright("Preparation done!"));
+        console.log(chalk.cyanBright("Performing variable and flag scans..."));
+
+        //create globs (empty content vfs)
+        let files: VirtualFileSystemInstance[] = cyanSafe.globs.Map((g: GlobSafe) => globFactory.GenerateFiles(g, g.root)).Flatten();
+
+        //remove package.lock
+        files = files.Where(f => { 
+            return VirtualFileSystemInstance.match(f, {
+                File: (file: FileSystemInstance) => {
+                    return !file.metadata.sourceAbsolutePath.includes("node_modules");
+                },
+                Folder: (folder: DirectorySystemInstance) => {
+                    return !folder.metadata.sourceAbsolutePath.includes("node_modules");
+                },
+                default: () => false
+            });
+        })
+        .Where(f => {
+            return VirtualFileSystemInstance.match(f, {
+                File: (file: FileSystemInstance) => {
+                    return (file.metadata.sourceAbsolutePath.ReplaceAll("\\\\", "/").split("/").Last()! !== "package-lock.json");
+                },
+                Folder: (folder: DirectorySystemInstance) => {
+                    return (folder.metadata.sourceAbsolutePath.ReplaceAll("\\\\", "/").split("/").Last()! !== "package-lock.json");
+                },
+                default: () => false
+            });
+        })
+        .Where(f => {
+            return VirtualFileSystemInstance.match(f, {
+                File: (file: FileSystemInstance) => {
+                    return (file.metadata.sourceAbsolutePath.ReplaceAll("\\\\", "/").split("/").Last()! !== "yarn.lock");
+
+                },
+                Folder: (folder: DirectorySystemInstance) => {
+                    return (folder.metadata.sourceAbsolutePath.ReplaceAll("\\\\", "/").split("/").Last()! !== "yarn.lock");
+                },
+                default: () => false
+            });
+        });
```

Comment:
@kirinnee Correct me if i'm wrong also, but I think this portion is not needed anymore (for now at least, these part may return in a plugin)

Cause these portion is for cleaning up the lock files and `node_modules` but in the new Cyanprint, by default we don't consider this and I think should be handled by a plugin?